rest-client/src/main/java/org/sagebionetworks/bridge/rest/RestUtils.java

```diff
@@ -198,20 +198,47 @@ public static String getUserAgent(ClientInfo info) {
         if (info == null) {
             return null;
         }
-        List<String> stanzas = Lists.newArrayListWithCapacity(3);
-        if (isNotBlank(info.getAppName()) && info.getAppVersion() != null) {
-            stanzas.add(String.format("%s/%s", info.getAppName(), info.getAppVersion()));
+        // Send what is available. The server can handle all meaningful combinations.
+        StringBuilder sb = new StringBuilder();
+        if (isNotBlank(info.getAppName()) || info.getAppVersion() != null) {
+            if (isNotBlank(info.getAppName())) {
+                sb.append(info.getAppName());
+            }
+            if (info.getAppVersion() != null) {
+                sb.append("/");
+                sb.append(info.getAppVersion());
+            }
```

Comment:
We should have unit tests for app name without version and app version without name.

Reply:
Added